torchgeo/models/cnnlstm.py

```diff
@@ -0,0 +1,378 @@
+                For classification:
+                    - 'outputs': Raw logits tensor (B, num_classes).
+                    - 'probs': Softmax probabilities tensor (B, num_classes).
+                For regression:
+                    - 'outputs': Regression predictions (B, output_dim).
+                    - 'probs': None.
+        """
+        # Apply pooling if input is sequence (B, T, input_dim)
+        if x.dim() == 3:
+            if self.pooling == 'last':
+                x = x[:, -1, :]
+            elif self.pooling == 'mean':
+                x = x.mean(dim=1)
+            elif self.pooling == 'max':
+                x = x.max(dim=1)[0]
+            elif self.pooling == 'attention':
+                attn_weights = torch.softmax(self.attention(x), dim=1)
+                x = (x * attn_weights).sum(dim=1)
+
+        x = self.dropout(x)
+        outputs = self.head(x)
+
+        if self.task_type == 'classification':
+            probs = torch.softmax(outputs, dim=-1)
+            return {'outputs': outputs, 'probs': probs}
+        else:  # regression
+            return {'outputs': outputs, 'probs': None}
+
+
+class CNNLSTM(nn.Module):
+    """Combined CNN + LSTM model with a unified task head.
+
+    Args:
+        cnn_backbone (CNN): CNN feature extractor module.
+        rnn_encoder (LSTM): LSTM sequence encoder module.
+        task_head (Task): Task head for classification or regression.
+
+    Raises:
+        ValueError: If the output dimension of CNN does not match input dimension of LSTM,
+                    or if the output dimension of LSTM does not match input dimension of task head.
+
+    Forward:
+        x (torch.Tensor): Input tensor, either
+            - (B, T, C, H, W) sequence of images, or
+            - (B, C, H, W) single image.
+        lengths (torch.Tensor, optional): Lengths of sequences for LSTM packing.
+
+    Returns:
+        Dict[str, Optional[torch.Tensor]]: Output dictionary from task head.
+    """
+
+    def __init__(self, cnn_backbone: CNN, rnn_encoder: LSTM, task_head: Task) -> None:
+        """Initialize CNN-LSTM model.
+
+        Args:
+            cnn_backbone: CNN feature extractor module.
+            rnn_encoder: LSTM sequence encoder module.
+            task_head: Task head for classification or regression.
+
+        Raises:
+            ValueError: If the output dimension of CNN does not match input dimension of LSTM,
+                        or if the output dimension of LSTM does not match input dimension of task head.
+        """
+        super().__init__()
+        self.cnn_backbone = cnn_backbone
+        self.rnn_encoder = rnn_encoder
+        self.task_head = task_head
+
+        # Validate CNN -> RNN dimensions
+        if cnn_backbone.output_dim != rnn_encoder.input_dim:
+            raise ValueError(
+                f'CNN output dim ({cnn_backbone.output_dim}) must match '
+                f'RNN input dim ({rnn_encoder.input_dim})'
+            )
```

Comment:
If there is only a single CNNLSTM class exposed and the user passes strings and numbers to that class, we won't need to check for this.